workspaces/optic-domain/src/streams/spec-events.ts

```diff
@@ -0,0 +1,25 @@
+import { Readable } from 'stream';
+import { takeWhile } from 'axax/esnext/takeWhile';
+
+import { fromJSONArray } from '../async-tools';
+
+type Event = any; // TODO change to event typing
+
+export function fromJSONStream(): (source: Readable) => AsyncIterable<Event> {
+  return fromJSONArray<Event>();
+}
+
+export function takeBatchesUntil(
+  batchCommitId: string
+): (iterable: AsyncIterable<Event>) => AsyncIterable<Event> {
+  let seenBatchCommitId = false;
+  return takeWhile((event) => {
+    if (seenBatchCommitId) {
+      return false;
+    }
+    if (event.BatchCommitEnded?.batchId === batchCommitId) {
+      seenBatchCommitId = true;
+    }
+    return true;
+  });
+}
```

Comment:
I realized my initial implementation was incorrect - we want to _include_ the batchCommit specified here, everything else should be ignored. This code now is correct (updated the test as well)